backend/src/apiserver/server/pipeline_server.go

```diff
@@ -184,6 +184,31 @@ func (s *PipelineServer) GetPipeline(ctx context.Context, request *api.GetPipeli
 	return ToApiPipeline(pipeline), nil
 }
 
+func (s *PipelineServer) GetPipelineByName(ctx context.Context, request *api.GetPipelineByNameRequest) (*api.Pipeline, error) {
+	if s.options.CollectMetrics {
+		getPipelineRequests.Inc()
+	}
+
+	//If namespace is "-" transform it to an emtpy string ""
+	if request.Namespace == model.NoNamespace {
+		request.Namespace = ""
+	}
```

Comment:
I am not in favor of this. Any other idea?

Reply:
How about assigning namespace to resourceAttributes only when it is multi user mode? 

Example: https://github.com/kubeflow/pipelines/blob/efd35e77e12c4b6d47ceabc8c64c3fc52672ee69/backend/src/apiserver/server/pipeline_server.go#L408

Reply:
done